Add adaptive_avg_pool2d and flatten into x86 Inductor Quantizer recipe

leslie-fang-intel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #114443
* __->__ #114442

**Summary**
Add adaptive_avg_pool2d and flatten into x86 Inductor Quantizer recipe

**Test Plan**
```
python -m pytest test_x86inductor_quantizer.py -k test_adaptive_avg_pool2d_recipe
python -m pytest test_x86inductor_quantizer.py -k test_flatten_recipe
```

leslie-fang-intel committed:
Add adaptive_avg_pool2d and flatten into x86 Inductor Quantizer recipe

[ghstack-poisoned]

jgong5 commented on torch/ao/quantization/quantizer/x86_inductor_quantizer.py:
```diff
@@ -68,6 +70,8 @@ class _X86InductorQuantizationAnnotation(QuantizationAnnotation):
     torch.ops.aten.max_pool2d.default,
     torch.ops.aten.cat.default,
     torch.ops.aten.avg_pool2d.default,
+    torch.ops.aten.adaptive_avg_pool2d.default,
+    torch.ops.aten.flatten.using_ints,
```
Is `int8_in_int8_out_ops_pt2e` also `quantizable_ops_pt2e`? If so, we can perhaps define `quantizable_ops_pt2e` with `int8_in_int8_out_ops_pt2e` plus additions?

leslie-fang-intel replied:
Thank you, yes, I agree. `int8_in_int8_out_ops_pt2e` should ultimately be a subset of `quantizable_ops_pt2e` (currently, these two sets are equal). Therefore, we can define `quantizable_ops_pt2e` by including `int8_in_int8_out_ops_pt2e` and additional operators.

leslie-fang-intel committed:
Update on "Add adaptive_avg_pool2d and flatten into x86 Inductor Quantizer recipe"


**Summary**
Add adaptive_avg_pool2d and flatten into x86 Inductor Quantizer recipe

**Test Plan**
```
python -m pytest test_x86inductor_quantizer.py -k test_adaptive_avg_pool2d_recipe
python -m pytest test_x86inductor_quantizer.py -k test_flatten_recipe
```


[ghstack-poisoned]

leslie-fang-intel commented:
@pytorchbot merge
